src/emulator/auth/operations.ts

```diff
@@ -881,18 +886,35 @@ export function setAccountInfoImpl(
   if (reqBody.oobCode) {
     const oob = state.validateOobCode(reqBody.oobCode);
     assert(oob, "INVALID_OOB_CODE");
-    if (oob.requestType !== "VERIFY_EMAIL") {
-      throw new NotImplementedError(oob.requestType);
-    }
-    state.deleteOobCode(reqBody.oobCode);
-
-    signInProvider = PROVIDER_PASSWORD;
-    const maybeUser = state.getUserByEmail(oob.email);
-    assert(maybeUser, "INVALID_OOB_CODE");
-    user = maybeUser;
-    updates.emailVerified = true;
-    if (oob.email !== user.email) {
-      updates.email = oob.email;
+    switch (oob.requestType) {
+      case "VERIFY_EMAIL": {
+        state.deleteOobCode(reqBody.oobCode);
+        signInProvider = PROVIDER_PASSWORD;
+        const maybeUser = state.getUserByEmail(oob.email);
+        assert(maybeUser, "INVALID_OOB_CODE");
+        user = maybeUser;
+        updates.emailVerified = true;
+        if (oob.email !== user.email) {
+          updates.email = oob.email;
+        }
+        break;
+      }
+      case "RECOVER_EMAIL": {
+        state.deleteOobCode(reqBody.oobCode);
+        // TODO: Ask if this is needed for RECOVER_EMAIL.
+        // signInProvider = PROVIDER_PASSWORD;
```

Comment:
AFAIU, setting the signInProvider is needed to issue tokens in the response. But it is not clear how that's used or when it is necessary.

Thanks!

Reply:
Please test this against production to see if a new token is indeed returned. If so, you can implement that by setting `signInProvider`.

Reply:
Will do. Thanks!

Reply:
I tested against prod. Tokens are not issued.